tfhe/src/core_crypto/fft_impl/fft64/crypto/bootstrap.rs

```diff
@@ -243,54 +244,61 @@ impl<'a> FourierLweBootstrapKeyView<'a> {
-                    &mut poly,
-                    MonomialDegree(monomial_degree),
-                )
+                let (mut tmp, _) = stack
+                    .rb_mut()
+                    .make_aligned_raw(poly.as_ref().len(), CACHELINE_ALIGN);
+
+                let mut tmp = Polynomial::from_container(&mut *tmp);
+                tmp.as_mut().copy_from_slice(poly.as_ref());
+                polynomial_wrapping_monic_monomial_div(&mut poly, &tmp, monomial_degree)
             });
 
         // We initialize the ct_0 used for the successive cmuxes
         let mut ct0 = lut;
+        let (mut ct1, mut stack) = stack.make_aligned_raw(ct0.as_ref().len(), CACHELINE_ALIGN);
+        let mut ct1 =
+            GlweCiphertextMutView::from_container(&mut *ct1, lut_poly_size, ciphertext_modulus);
 
         for (lwe_mask_element, bootstrap_key_ggsw) in izip!(lwe_mask.iter(), self.into_ggsw_iter())
         {
             if *lwe_mask_element != Scalar::ZERO {
-                let stack = stack.rb_mut();
-                // We copy ct_0 to ct_1
-                let (mut ct1, stack) =
-                    stack.collect_aligned(CACHELINE_ALIGN, ct0.as_ref().iter().copied());
-                let mut ct1 = GlweCiphertextMutView::from_container(
-                    &mut *ct1,
+                let monomial_degree = MonomialDegree(pbs_modulus_switch(
+                    *lwe_mask_element,
                     lut_poly_size,
-                    ciphertext_modulus,
-                );
-
-                // We rotate ct_1 by performing ct_1 <- ct_1 * X^{a_hat}
-                for mut poly in ct1.as_mut_polynomial_list().iter_mut() {
-                    polynomial_wrapping_monic_monomial_mul_assign(
-                        &mut poly,
-                        MonomialDegree(pbs_modulus_switch(
-                            *lwe_mask_element,
-                            lut_poly_size,
-                            ModulusSwitchOffset(0),
-                            LutCountLog(0),
-                        )),
+                    ModulusSwitchOffset(0),
+                    LutCountLog(0),
+                ));
+
+                // We rotate ct_1 by performing ct_1 <- (ct_0 * X^{a_hat}) - ct_0
+                for (mut ct1_poly, ct0_poly) in zip(
+                    ct1.as_mut_polynomial_list().iter_mut(),
+                    ct0.as_polynomial_list().iter(),
+                ) {
+                    polynomial_wrapping_monic_monomial_mul_and_subtract(
+                        &mut ct1_poly,
+                        &ct0_poly,
+                        monomial_degree,
                     );
                 }
 
-                // ct1 is re-created each loop it can be moved, ct0 is already a view, but
                 // as_mut_view is required to keep borrow rules consistent
-                cmux(ct0.as_mut_view(), ct1, bootstrap_key_ggsw, fft, stack);
+                add_external_product_assign(
```

Comment:
small comment here to explain that we are effectively inlining/doing a precomputation that means the external product is computing the cmux as expected